plugins/hls-rename-plugin/src/Ide/Plugin/Rename.hs

```diff
@@ -74,7 +78,10 @@ renameProvider state pluginId (RenameParams (TextDocumentIdentifier uri) pos _pr
            See the `IndirectPuns` test for an example. -}
         indirectOldNames <- concat . filter ((>1) . Prelude.length) <$>
             mapM (uncurry (getNamesAtPos state) . locToFilePos) directRefs
-        let oldNames = indirectOldNames ++ directOldNames
+        let oldNames = (filter matchesDirect indirectOldNames) ++ directOldNames
+            matchesDirect n = occNameFS (nameOccName n) `elem` directFS
+              where
+                directFS = map (occNameFS. nameOccName) directOldNames
```

Comment:
Why is this necessary? Is there a case that we were missing?

The only case I am aware of for indirect names is with record field puns, but these should always "match" the direct names since they are, well, puns.

Looks good otherwise - many thanks

Reply:
If there are implicit dictionary names (eg `$fShow` for code like `show 1`) then we don't want to attempt to rename those.

Reply:
You should mention that you need to filter stuff out since GHC-9.4 because now there are implicit dictionary names like `$fShow` which we don't want to attempt to rename.